irods/library/irods_move

```diff
@@ -1,142 +1,177 @@
+
+RETURN = '''
+exc:
+    description:
+        - type of last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+exc_msg:
+    description:
+        - message of the last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+'''
+
+
+class IRODSMoveModule:
+    """
+    Module class
+    """
+
+    def __init__(self):
+        """
+        Initialize module
+        """
+
+        self.module_args = dict(
+            source=dict(type="str", required=True),
+            destination=dict(type="str", required=True),
+
+            host=dict(type="str", required=True),
+            port=dict(type="int", required=True),
+            admin_user=dict(type="str", no_log=True, required=True),
+            admin_password=dict(type="str", no_log=True, required=True),
+            zone=dict(type="str", required=True),
+        )
+
+        self.result = dict(
+            changed=False,
+            exc='',
+            exc_msg=''
+        )
+
+        self.module = AnsibleModule(
+            argument_spec=self.module_args,
+            supports_check_mode=True
+        )
+
+        self.ssl_context = ssl.create_default_context(
+            purpose=ssl.Purpose.SERVER_AUTH, cafile=None, capath=None,
+            cadata=None)
+
+        self.ssl_settings = {"ssl_context": self.ssl_context}
+
+        self.session = None
+
+    def init_session(self):
+        try:
+            self.session = iRODSSession(
+                source=self.module.params["source"],
+                destination=self.module.params["destination"],
+                host=self.module.params["host"],
+                port=self.module.params["port"],
+                user=self.module.params["admin_user"],
+                password=self.module.params["admin_password"],
+                zone=self.module.params["zone"],
+                **self.ssl_settings)
+
+        except Exception as exc:
+            self.result["exc"] = exc
+            self.result["exc_message"] = "Unable to connect to iRODS."
+            self._fail(self.result["exc_message"])
+
+    def run_module(self):
+        if self.module.check_mode:
+            self.module.exit_json(**self.result)
+
+        self.init_session()
+
+        try:
+            self.rename()
+            self.module.exit_json(**self.result)
+        except Exception as exc:
+            self.result["exc"] = type(exc).__name__
+            self.result["exc_msg"] = str(exc)
+            self._fail("Unable to move irods object/collection {}".format(
+                self.module.params["source"]), exc)
+
+    def rename(self):
+        # rename collection
+        path = self.module.params["source"]
+        new_path = self.module.params["destination"]
+        path_exists = self.session.collections.exists(path)
+        new_path_exists = self.session.collections.exists(new_path)
+
+        if path_exists is True and new_path_exists is False:
```

Comment:
There are four cases:

1) source and destination exist  -- This is an error, won't overwrite existing collection or data object
2) source exists but destination doesn't -- This is when collections.move is called and change = True
3) source doesn't exist but destination does -- This is not an error, we silently assume the move already happened, change = False; we make this assumption to support idempotency.
4) neither source nor destination exist. -- This is an error, source doesn't exist

You only need to test for 3).  collections.move will handle cases 1), 2), and 4) correctly.